server/base/src/main/java/org/apache/accumulo/server/master/recovery/RecoveryPath.java

```diff
@@ -42,7 +48,8 @@ public static Path getRecoveryPath(Path walPath) {
       // drop wal
       walPath = walPath.getParent();
 
-      walPath = new Path(walPath, FileType.RECOVERY.getDirectory());
+      walPath = new Path(walPath,
+          FileType.RECOVERY.getDirectory() + '/' + context.getUniqueNameAllocator().getNextName());
```

Comment:
It would be good to log a message about the name that was chosen, before being used here, as a debug message, so that a an admin can trace the specific recovery directory back to a specific server from the server logs if necessary.

Instead of using `'/'`, you can use `'-'` so that way the directories look like `recovery-<uniquePart>`.